src/main/java/net/minecraftforge/event/entity/player/PlayerInteractEvent.java

```diff
@@ -198,6 +218,35 @@ public void setCanceled(boolean canceled)
         }
     }
 
+    /**
+     * This event is fired on both sides whenever the player would be able to use an item, after clicking on a block.
+     * This event is called when {@link net.minecraft.item.Item#onItemUse} would be called.
+     * Canceling the event will cause the item to not be used.
+     *
+     * Note that handling things differently on the client vs server may cause desynchronizations!
+     */
+    @Cancelable
+    public static class OnItemUse extends PlayerInteractEvent
+    {
+        private final BlockRayTraceResult res;
+
+        public OnItemUse(PlayerEntity player, Hand hand, BlockPos pos, BlockRayTraceResult res) {
+            super(player, hand, pos, res.getFace());
+            this.res = res;
+        }
+
+        /**
+         * @return The ray trace result from clicking on this block.  May be null, if called from mod code using the old constructor.
+         * Mods should be phasing away from the old constructor, and the nullable annotation will be removed in a future version.
+         */
+        @Nullable
```

Comment:
This class is new, and you call a method on the `BlockRayTraceResult` in the constructor, so this has to be `Nonnull`

Reply:
Yep, that's what I get for copying the RightClickBlock class